lib/Web/Server.hs

```diff
@@ -42,6 +47,89 @@ url = BaseUrl Http "127.0.0.1" 8200 ""
 mount :: Text
 mount = "secret"
 
+data CofferServerError = CofferServerError
+  { cseError :: Text
+  , cseCode :: Int
+  }
+  deriving stock (Show, Generic)
+deriveJSON (aesonPrefix pascalCase) ''CofferServerError
+
+cofferServerErrorToBytes :: Text -> Int -> Bytes.Lazy.ByteString
+cofferServerErrorToBytes msg code =
+  CofferServerError msg code
+    & toEncoding
+    & encodingToLazyByteString
+
+throwViewError :: ViewResult -> Handler a
+throwViewError = \case
+  VRDirectory{} -> throwError err500 { errBody = "Expected view error, got VRDirectory." }
+  VREntry{} -> throwError err500 { errBody = "Expected view error, got VREntry." }
+  VRField{} -> throwError err500 { errBody = "Expected view error, got VRField." }
+  res@VRPathNotFound{} -> throwError err404 { errBody = cofferServerErrorToBytes (pretty res) 100 }
+  res@VRDirectoryNoFieldMatch{} -> throwError err404 { errBody = cofferServerErrorToBytes (pretty res) 101 }
+  res@VREntryNoFieldMatch{} -> throwError err404 { errBody = cofferServerErrorToBytes (pretty res) 102 }
+
+throwCreateError :: CreateResult -> Handler a
+throwCreateError = \case
+  CRSuccess{} -> throwError err500 { errBody = "Expected create error, got CRSuccess." }
+  CRCreateError createError ->
+    case createError of
+      res@CEParentDirectoryIsEntry{} -> throwError err409 { errBody = cofferServerErrorToBytes (pretty res) 200 }
+      res@CEDestinationIsDirectory{} -> throwError err409 { errBody = cofferServerErrorToBytes (pretty res) 201 }
+      res@CEEntryAlreadyExists{} -> throwError err409 { errBody = cofferServerErrorToBytes (pretty res) 202 }
+
+throwSetFieldError :: SetFieldResult -> Handler a
+throwSetFieldError = \case
+  SFRSuccess{} -> throwError err500 { errBody = "Expected set-field error, got SFRSuccess." }
+  res@SFREntryNotFound{} -> throwError err404 { errBody = cofferServerErrorToBytes (pretty res) 300 }
+  res@SFRMissingFieldContents{} -> throwError err400 { errBody = cofferServerErrorToBytes (pretty res) 301 }
+
+throwDeleteFieldError :: DeleteFieldResult -> Handler a
+throwDeleteFieldError = \case
+  DFRSuccess{} -> throwError err500 { errBody = "Expected delete-field error, got DFRSuccess." }
+  res@DFREntryNotFound{} -> throwError err404 { errBody = cofferServerErrorToBytes (pretty res) 400 }
+  res@DFRFieldNotFound{} -> throwError err404 { errBody = cofferServerErrorToBytes (pretty res) 401 }
+
+throwCopyOrRenameError :: Bool -> CopyResult -> Handler a
```

Comment:
Same as with `buildCopyOrRenameResult`

Reply:
I think I should do the same with `copyOrRenameHandler`, not with this one